cmd/autoscaler/main.go

```diff
@@ -151,7 +151,7 @@ func main() {
 
 	// accept is the func to call when this pod owns the Revision for this StatMessage.
 	accept := func(sm asmetrics.StatMessage) {
-		collector.Record(sm.Key, time.Now(), sm.Stat)
+		collector.Record(sm.Key, time.Unix(sm.Stat.Timestamp, 0), sm.Stat)
```

Comment:
one downside to this that just occurred to me is I think it means we can have events arriving out of order (some from the old autoscaler with a lag, some arriving directly at the new autoscaler), in which case IIUC we could end up calling [float64buckets.Record](https://github.com/knative/serving/blob/master/pkg/autoscaler/aggregation/bucketing.go#L155) with non-monotonically increasing values for the `now` parameter. Looking at the tests for that and scanning the code I'm not sure we've ever really considered that case (e.g. I think if we get old data we'll [set lastWrite to the earlier time](https://github.com/knative/serving/blob/master/pkg/autoscaler/aggregation/bucketing.go#L188) even though we've written later data). Is that going to be a problem here?

Reply:
That was my comment about rewriting history in the issue :)

Reply:
Forgot about that. Yeah it will be a problem i think. Will take a look,

Reply:
Fixed.